### PR TITLE
Add regression test for subs

### LINKED ISSUE #11746: undesired (wrong) substition behavior in sympy?
I'm getting this undesired behavior in sympy

In [1]: x = sy.var('x', real=True)

In [2]: (1/x).subs(x**2,1)
Out[2]: 1

Clearly, this should not be the case. Out[2] should be 1/x. Am I missing something or is this a bug?

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

Fixes #11746

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by namannimmo10** [comment]
@oscarbenjamin If any changes are still required, please suggest, otherwise, I think the request is ready to be merged.

File: sympy/core/tests/test_subs.py [hunk 1]
```diff
@@ -831,3 +831,12 @@ def test_issue_6976():
     assert (x**4 + x**3 + x**2 + x + sqrt(x)).subs(x**2, y) == \
         y**Rational(1, 4) + y**Rational(3, 2) + sqrt(y) + y**2 + y
     assert x.subs(x**3, y) == y**Rational(1, 3)
+
+
+def test_issue_11746():
+    x, y = symbols('x y')
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_subs.py]: I think these symbols are defined at the top of the file so that they don't need to be defined in each test

Comment (namannimmo10) [reply sympy/core/tests/test_subs.py]: Sure, I made the change.  Thanks for the review!

File: sympy/solvers/diophantine.py [hunk 2]
```diff
@@ -1004,7 +1004,7 @@ def _diop_quadratic(var, coeff, t):
                 for z0 in range(0, abs(_c)):
                     # Check if the coefficients of y and x obtained are integers or not
                     if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
-                            divisible(e*sqc**g*z0**2 + E*z0 + e*sqc*F, _c)):
+                            divisible(e*sqc*g*z0**2 + E*z0 + e*sqc*F, _c)):
```
Comment (smichr) [inline sympy/solvers/diophantine.py]: This was a good catch. If it hasn't been fixed in master and no tests failed, the following test should be added (and I confirmed that the solution satisfies, in general, the original equation):
```python
assert diophantine(8*x**2 - 24*x*y + 18*y**2 + 5*x + 7*y + 16) == {
    (-174*t**2 + 41*t - 4, -116*t**2 + 37*t - 4),
    (-174*t**2 + 17*t - 2, -116*t**2 + 21*t - 2)}
```
That test fails in 1.5.

Comment (smichr) [reply sympy/solvers/diophantine.py]: cf https://github.com/sympy/sympy/issues/18156

Comment (namannimmo10) [reply sympy/solvers/diophantine.py]: That typo has been already fixed in [#18130](https://github.com/sympy/sympy/pull/18130#issue-356831749). My sympy version is '1.6.dev'. That test is successful in that version.

**Comment by namannimmo10** [comment]
@oscarbenjamin is it ready to be merged?

**Comment by oscarbenjamin** [comment]
The PR contains an unrelated commit for diophantine. Can you remove that?

**Comment by namannimmo10** [comment]
Ok, I'll do that. But does it really matter cause that change has been done already in the master branch?

**Comment by oscarbenjamin** [comment]
I've squashed this to a single commit

**Comment by namannimmo10** [comment]
it's ready now. Please merge this PR.

**Pull request merged**